internal/controller/utils.go

```diff
@@ -16,6 +17,35 @@ func updateDeploymentAnnotations(deployment *appsv1.Deployment, annotations map[
 	}
 }
 
+// syncDeploymentReplicas synchronizes the replicas field between two Deployment objects.
+func syncDeploymentReplicas(a, b *appsv1.Deployment) bool {
+	if a == nil || b == nil {
+		return false
+	}
+
+	if *a.Spec.Replicas != *b.Spec.Replicas {
+		a.Spec.Replicas = b.Spec.Replicas
+		return true
+	}
+
+	return false
+}
+
+func syncDeploymentResources(a, b *appsv1.Deployment) bool {
+	if a == nil || b == nil {
+		return false
+	}
+
+	existingResources := &a.Spec.Template.Spec.Containers[0].Resources
+	desiredResources := b.Spec.Template.Spec.Containers[0].Resources
+	if !apiequality.Semantic.DeepEqual(*existingResources, desiredResources) {
+		*existingResources = desiredResources
+		return true
+	}
```

Comment:
We can also move this DeepEqual checks to its own function. We may want to reuse for other deployment like Redis in the future.

Reply:
just a nit - `func deploymentEquals(current, new *appsv1.Deployment) bool`  , we may be able to use for redis deployment check.